a simpler maps service

toddjordan opened the pull request:
Ok, this is a simpler version of the maps service.  It is simply creating a map and displaying it based on location.  Its also caching for later, leveraging the singleton nature of services (for use when we implement a details/edit view)

toddjordan commented on app/index.html:
```diff
@@ -19,6 +19,7 @@
 
     <script src="assets/vendor.js"></script>
     <script src="assets/super-rentals.js"></script>
+    <script src="https://maps.googleapis.com/maps/api/js?v=3.23"></script>
```
For simplicity, I'm globally importing the maps script.  Open to better ways with maps.

knownasilya replied:
This is the best way, apart from using the inline-content addon, which is probably too much here.

toddjordan commented on app/utils/google-maps.js:
```diff
@@ -0,0 +1,32 @@
+/*globals google*/
+import Ember from 'ember';
+
+let geocoder = new google.maps.Geocoder();
```
I tried to extract the google dependencies into a utils file, so I can more easily stub them for tests

toddjordan commented on tests/integration/components/location-map-test.js:
```diff
@@ -0,0 +1,24 @@
+  integration: true
+});
+
+test('it renders', function(assert) {
```
This is just the default generated, I plan to add a few tests around validating that the maps get rendered.

toddjordan commented on tests/unit/services/maps-test.js:
```diff
@@ -0,0 +1,14 @@
+  // needs: ['service:foo']
+});
+
+test('should fetch a map element based on location', function(assert) {
```
This test is just verifying that the element gets created and properly returned.  My plan is to stub the google util and verify that create is called when not in cache, and skipped when it is.  I also will make sure the pin is called.  This one is actually calling through to google.  not sure we want to do this.

toddjordan commented on tests/unit/utils/google-maps-test.js:
```diff
@@ -0,0 +1,10 @@
+import googleMaps from 'super-rentals/utils/google-maps';
```
Not sure I want to unit test this because of the google network requests . I'll do some research.

toddjordan commented on app/components/location-map.js:
```diff
@@ -0,0 +1,12 @@
+
+  didInsertElement() {
+    this._super(...arguments);
+    Ember.Logger.debug('didInsertElement');
```
I'll delete all the logger statements before merge

knownasilya commented on app/components/location-map.js:
```diff
@@ -0,0 +1,12 @@
+  didInsertElement() {
+    this._super(...arguments);
+    Ember.Logger.debug('didInsertElement');
+    this.$('.map-container').append(this.get('maps').getMapElement(this.get('location')));
```
I'd be more explicit about naming the service, maps => mapService just so it's clear. Just a thought.

acorncom replied:
Given that idiomatic Ember tends to use the service name like we've got here, I would be inclined to leave as is (but maybe add a code comment explaining what is going on above this line?)

knownasilya replied:
Or do `var mapService = this.get('maps')` keeping the inject idiomatic?

Really, it would benefit from a few variables here anyway.

toddjordan commented:
Updated with tests!
Added unit tests for the maps service and an integration test for the component.
Also changed the approach for including the maps api to match what I originally added for the tutorial (using curl to download it into the vendor dir)
